lib/facade/MonitoringFacade.ts

```diff
@@ -138,8 +140,13 @@ export interface MonitoringFacadeProps {
 export class MonitoringFacade extends MonitoringScope {
   protected readonly metricFactoryDefaults: MetricFactoryDefaults;
   protected readonly alarmFactoryDefaults: AlarmFactoryDefaults;
-  protected readonly dashboardFactory?: IDashboardFactory;
-  protected readonly createdSegments: IDashboardSegment[];
+  protected readonly dashboardFactory?:
+    | IDashboardFactory
+    | IDynamicDashboardFactory;
+  protected readonly createdSegments: (
+    | IDashboardSegment
+    | IDynamicDashboardSegment
+  )[];
```

Comment:
These would ideally just be a single base interface that the others also implement, although it's unfortunate that the existing ones are already the generic names but tied to specific implementations.

Reply:
I've addressed this concern by modifying the default dashboard factory to also be an implementation of the dynamic dashboard factory interface.